tfjs-backend-webgpu/src/kernels/matmul_packed_vec4_webgpu.ts

```diff
@@ -110,6 +110,163 @@ export function makeMatMulPackedVec4Source(workPerThread: number[]): string {
   `;
 }
 
+function getSharedArray1DCodeWgsl() {
+  return `
+  var<workgroup> mm_Asub : array<vec4<f32>, 1024>;`;
+}
+
+function getSharedArray2DCodeWgsl() {
+  return `
+  var<workgroup> mm_Asub : array<array<vec4<f32>, 16>, 64>;
+  var<workgroup> mm_Bsub : array<array<vec4<f32>, 16>, 64>;`;
+}
+
+function getA1DCodeWgsl() {
+  return `
+  let index : u32 = inputRow * TileInner / ColPerThread + inputCol;
+  mm_Asub[index] = mm_readA(globalRow + innerRow, globalColA, globalId);`;
+}
+
+function getA2DCodeWgsl() {
+  return `
+  mm_Asub[inputRow][inputCol] = mm_readA(globalRow + innerRow, globalColA, globalId);`;
+}
+
+function getB1DCodeWgsl() {
+  return `
+  let index : u32 = inputRow * TileBOuter / ColPerThread + inputCol;
+  mm_Bsub[index] = mm_readB(t * TileInner + inputRow, globalCol);`;
+}
+
+function getB2DCodeWgsl() {
+  return `
+  mm_Bsub[inputRow][inputCol] = mm_readB(t * TileInner + inputRow, globalCol);`;
+}
+
+function getCompute1DCodeWgsl() {
+  return `
+  // Compute acc values for a single thread.
+  for (var k : u32 = 0u; k < TileInner / ColPerThread; k = k + 1u) {
+      BCached[0] = mm_Bsub[(k * ColPerThread) * (TileBOuter / ColPerThread) + tileCol];
```

Comment:
There is no `mm_Bsub`. Please make sure your shader is tested.

Reply:
done